sys/sched.c

```diff
@@ -70,20 +70,45 @@ static void sched_set_priority(thread_t *td, td_prio_t prio) {
   }
 }
 
+void sched_set_prio(thread_t *td, td_prio_t prio) {
+  assert(spin_owned(td->td_spin));
+
+  td->td_base_prio = prio;
+
+  /* If thread is borrowing priority, don't lower its active priority. */
+  if (td->td_flags & TDF_BORROWING && td->td_prio > prio)
+    return;
+
+  td_prio_t oldprio = td->td_prio;
+  sched_set_active_prio(td, prio);
+
+  /* If thread is locked on a turnstile, let the turnstile adjust
+   * thread's position on turnstile's \a ts_blocked list. */
+
+  /* These changes are waiting for turnstile pull request because
+   * now we don't have TDS_LOCKED and turnstile_adjust. */
+  // TO BE UNCOMMENTED
```

Comment:
@melouber @cahirwpz, perhaps it would make more sense if we introduced minimal changes required for the commented code (TDS_LOCKED flag and empty procedure turnstile_adjust)?

Reply:
Please remove all code (and comments) that are not relevant to this change.